ImageData/Hdf5Loader.h

```diff
@@ -327,6 +343,8 @@ std::map<CARTA::StatsType, std::vector<double>>* Hdf5Loader::GetRegionSpectralDa
             std::piecewise_construct, std::forward_as_tuple(region_id, stokes), std::forward_as_tuple(origin, mask->shape(), num_z));
         recalculate = true;
     } else if (!_region_stats[region_stats_id].IsValid(origin, mask->shape())) { // region stats expired
```

Comment:
One more thing: because we are now updating these stats in the loader state incrementally, and may abort the calculation before it's complete, we need to add a flag to the state which indicates that the calculation completed -- so that if we try to calculate stats for the same region later we don't think that the cached stats are valid if they were not fully calculated.

Reply:
If we want to be really clever, we can save the actual position to which we previously calculated, use a separate check, and resume from there instead of recalculating from scratch.